package.json

```diff
@@ -30,6 +30,31 @@
     "./src/*.js": "./src/*.js"
   },
   "main": "dist/jquery.js",
+  "scripts": {
+    "babel:tests": "babel test/data/core/jquery-iterability-transpiled-es6.js --out-file test/data/core/jquery-iterability-transpiled.js",
+    "build": "node ./build/command.js",
+    "build-all-variants": "node -e \"require('./build/tasks/build.js').buildDefaultFiles()\"",
+    "jenkins": "npm run test:browserless",
+    "lint:dev": "eslint --cache .",
+    "lint:json": "jsonlint --quiet package.json",
+    "lint": "concurrently -r \"npm:lint:dev\" \"npm:lint:json\"",
+    "prepare": "husky install",
+    "pretest": "npm run qunit-fixture && npm run babel:tests",
+    "qunit-fixture": "node build/tasks/qunit-fixture.js",
+    "start": "node -e \"require('./build/tasks/build.js').buildDefaultFiles({ watch: true })\"",
+    "test:browserless": "npm run test:node_smoke_tests && npm run test:promises_aplus && npm run test:jsdom",
+    "test:browser": "grunt karma:main",
+    "test:esmodules": "grunt karma:esmodules",
+    "test:jsdom": "grunt test:jsdom",
+    "test:no-deprecated": "npm run build -- -e deprecated && npm run test:browser",
+    "test:selector-native": "npm run build -- -e selector && npm run test:browser",
+    "test:slim": "npm run build -- --slim && npm run test:browser",
```

Comment:
It's a bit weird that some of the `test:*` scripts do run the build and others do not. Previously it was more consistent - all of them run the build. Perhaps you can rename some of those to make it less confusing?

BTW, it's a common pattern that `npm it` (or, in its longer form, `npm install && npm test`) should work and test the package so if tests need a build, `npm test` should also build.

Reply:
I didn't like that `npm test` was building the same thing multiple times, so I've removed all the duplicates. However, I agree that `npm install && npm test` should work without any other commands, but I think it already does. Here's the order of operations for `npm it` as I've tried to write it:

1. npm install
2. npm test triggers the `pretest` script automatically, which builds the qunit fixture and transpiles the babel script.
3. npm test is run, which does the following scripts in order

1. `npm run build-all-variants` - builds all the normal jQuery files, including the new module.js files 
2. `npm run lint` - lints the repo, including the just-built dist files
3. `npm run test:browserless` - runs node smoke tests, promises tests, and jsdom tests, which requires all built files
4. `npm run test:browser` - main karma tests, which requires files already built with `build-all-variants`
5. `npm run test:esmodules` - same as (4), but the modules files instead
6. `npm run test:slim` - runs main karma tests against the slim build, which is built in the `test:slim` command so that jquery.[min].js is the slim build (not just jquery.slim.js)
7. `npm run test:no-deprecated` - runs main karma tests against a custom build of jquery.[min].js that removes the deprecated module
8. `npm run test:selector-native` - runs main karma tests against a custom build that replaces the selector engine with selector-native

I think this all works fine without further building.

Reply:
With the proposed setup, `npm run test:browser && npm run test:slim` may run tests against the regular build & the slim build... or it may not use the regular build - who knows what's in the built file at that point. If you run `npm run test:slim && npm run test:browser` then you'll run tests against the slim build twice.

I don't like this inconsistency, it makes it hard to remember what does what and we will need to run these npm scripts manually during local development a lot.

I know `npm test` builds jQuery multiple times on `main` but I left it that way to avoid the above issues. The full `npm test` is not run on CI anywhere anyway so this was just for running tests locally. The build, even repeated a few times, is still a tiny fraction of time compared to all the tests run.

I'm fine with removing these multiple re-builds but then I'd definitely want more consistency in npm scripts, the current proposed setup doesn't fulfill that desire. The mental overhead is real, I'd rather not have to remember all these inconsistencies.

Reply:
I see your point. That's just not really been my experience. IMO, `npm test` is the only command we need to worry about when it comes to running tests out of context. The others are used individually when debugging their respective builds, but one at a time. In other words, I don't see a practical reason for needing `npm run test:slim && npm run test:browser` to work. I'm either debugging the slim build or the main one, and I'd say 99% of the time it's just the main one with a watch task, which is why I find the extra builds superfluous.

In fact, before this, I have found myself editing the build out of npm scripts locally. I found it easier than trying to remember which grunt test command it was. I'm not saying I can't be won over, but I can say I think the previous scripts have wasted more time than just the extra time the builds take during the main `test` script.

That said, if your personal experience has led you to wanting to add the build to the test scripts, let's braintstorm.

Reply:
I thought about this more and tested how long the build takes. It's fine. We'll just have some extra builds. I think I prefer that to more npm scripts, which was the only other option I could think of. 

Also, to that end, I've consolidated the `node_smoke_tests` scripts into one and moved the file combinations to JS. I think that will be better for your upcoming PR as well.